components/org.wso2.carbon.identity.application.authenticator.fido2/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/core/WebAuthnService.java

```diff
@@ -688,15 +701,46 @@ private static ByteArray generateRandom() {
         return new ByteArray(bytes);
     }
 
-    private StartRegistrationOptions buildStartRegistrationOptions(User user) {
+    private StartRegistrationOptions buildStartRegistrationOptions(User user, boolean requireResidentKey)
+            throws FIDO2AuthenticatorServerException {
 
         return StartRegistrationOptions.builder()
-                .user(buildUserIdentity(user)).build();
+                .user(buildUserIdentity(user))
+                .timeout(Integer.parseInt(userResponseTimeout))
+                .authenticatorSelection(buildAuthenticatorSelection(requireResidentKey))
+                .extensions(RegistrationExtensionInputs.builder().build())
+                .build();
+    }
+
+    private AuthenticatorSelectionCriteria buildAuthenticatorSelection(boolean requireResidentKey) {
+
+        return AuthenticatorSelectionCriteria.builder()
+                .requireResidentKey(requireResidentKey)
+                .userVerification(PREFERRED)
+                .build();
+    }
+
+    private String getUserDisplayName(User user) throws FIDO2AuthenticatorServerException {
+
+        String displayName = null;
+        String formattedNameClaimURL = "http://wso2.org/claims/formattedName";
```

Comment:
It would be better to have `formattedNameClaimURL` as a private static final String and use it in [1] and [2].

[1] https://github.com/wso2-extensions/identity-local-auth-fido/pull/58/files#diff-079bb238e9482998b1b65e3af34ac2c904a418c02c078166daf48a852ec65407R729
[2] https://github.com/wso2-extensions/identity-local-auth-fido/pull/58/files#diff-079bb238e9482998b1b65e3af34ac2c904a418c02c078166daf48a852ec65407R731

Reply:
Addressed with https://github.com/wso2-extensions/identity-local-auth-fido/pull/58/commits/1a8a836cb37d9cb3e595db5906f4cf9359a4b6ce